src/Form/BasicOverview.php

```diff
@@ -0,0 +1,281 @@
+   *
+   * @var \Drupal\Core\KeyValueStore\KeyValueStoreExpirableInterface
+   */
+  protected $keyValueExpirable;
+
+  /**
+   * The module installer.
+   *
+   * @var \Drupal\Core\Extension\ModuleInstallerInterface
+   */
+  protected $moduleInstaller;
+
+  /**
+   * The permission handler.
+   *
+   * @var \Drupal\user\PermissionHandlerInterface
+   */
+  protected $permissionHandler;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('module_handler'),
+      $container->get('module_installer'),
+      $container->get('keyvalue.expirable')->get('module_list'),
+      $container->get('access_manager'),
+      $container->get('current_user'),
+      $container->get('user.permissions')
+    );
+  }
+
+  /**
+   * Constructs a ModulesListForm object.
+   *
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module handler.
+   * @param \Drupal\Core\Extension\ModuleInstallerInterface $module_installer
+   *   The module installer.
+   * @param \Drupal\Core\KeyValueStore\KeyValueStoreExpirableInterface $key_value_expirable
+   *   The key value expirable factory.
+   * @param \Drupal\Core\Access\AccessManagerInterface $access_manager
+   *   Access manager.
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user.
+   * @param \Drupal\user\PermissionHandlerInterface $permission_handler
+   *   The permission handler.
+   */
+  public function __construct(ModuleHandlerInterface $module_handler, ModuleInstallerInterface $module_installer, KeyValueStoreExpirableInterface $key_value_expirable, AccessManagerInterface $access_manager, AccountInterface $current_user, PermissionHandlerInterface $permission_handler) {
+    $this->moduleHandler = $module_handler;
+    $this->moduleInstaller = $module_installer;
+    $this->keyValueExpirable = $key_value_expirable;
+    $this->accessManager = $access_manager;
+    $this->currentUser = $current_user;
+    $this->permissionHandler = $permission_handler;
+    $perms = $this->permissionHandler->getPermissions();
+    $search_config_permissions = array_filter($perms, function($v) {
+      return $v['provider'] == 'tfa';
+    });
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'tfa_basic_base_overview';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, UserInterface $user = NULL) {
+    $output['info'] = array(
+      '#type'   => 'markup',
+      '#markup' => '<p>' . t('Two-factor authentication (TFA) provides additional security for your account. With TFA enabled, you log in to the site with a verification code in addition to your username and password.') . '</p>',
+    );
+    // $form_state['storage']['account'] = $user;.
+    $configuration = \Drupal::config('tfa.settings')->getRawData();
+    $user_tfa      = tfa_get_tfa_data($user->id());
```

Comment:
Yes I am doubtful about this also. This is currently defined in the tfa.module file.